ODLA/platforms/dnnl/odla_dnnl.cc

```diff
@@ -1360,116 +1406,86 @@ odla_value odla_ReduceMean(odla_value input, odla_size_t num_of_axes,
-                   odla_value_shape output_dims, const odla_value_id id) {
-  const auto& lhs_dims = lhs->shape;
-  const auto& rhs_dims = rhs->shape;
-  auto dt = lhs->mem.get_desc().data_type();
-  long M = output_dims.dims[0], N = output_dims.dims[1],
-       K = !transpose_rhs ? rhs_dims.dims[0] : rhs_dims.dims[1];
-  long lda = transpose_lhs ? M : K;
-  long ldb = transpose_rhs ? K : N;
-  long ldc = N;
-  dnnl::memory::desc lhs_md(
-      {M, K}, dt,
-      transpose_lhs ? dnnl::memory::dims{1, lda} : dnnl::memory::dims{lda, 1});
-
-  dnnl::memory::desc rhs_md(
-      {K, N}, dt,
-      transpose_rhs ? dnnl::memory::dims{1, ldb} : dnnl::memory::dims{ldb, 1});
-
-  dnnl::memory::desc ret_md({M, N}, dt, {ldc, 1});
-  auto ret_mem = dnnl::memory(ret_md, g_comp->eng);
-  bool is_elements_add = false;
-  if (bias) {
-    auto bias_elements = GetTotalElements(bias->shape);
-    if (bias_elements == N) {
-      dnnl::memory::desc bias_md({1, N}, dt, dnnl::memory::format_tag::ab);
-      auto bias_mem =
-          dnnl::memory(bias_md, g_comp->eng, bias->mem.get_data_handle());
-      dnnl::matmul::desc md(lhs_md, rhs_md, bias_md, ret_md);
-      dnnl::matmul::primitive_desc pd(md, g_comp->eng);
-      dnnl::primitive prim = dnnl::matmul(pd);
-      add_op(prim, {{DNNL_ARG_SRC, lhs->mem},
-                    {DNNL_ARG_WEIGHTS, rhs->mem},
-                    {DNNL_ARG_BIAS, bias_mem},
-                    {DNNL_ARG_DST, ret_mem}});
-    } else if (bias_elements == 1) {
-      dnnl::post_ops ops;
-      dnnl::primitive_attr gemm_attr;
-      float beta = ((float*)bias->mem.get_data_handle())[0];
-      ops.append_eltwise(1.f, dnnl::algorithm::eltwise_linear, 0.f, beta);
-      gemm_attr.set_post_ops(ops);
-      dnnl::matmul::desc md(lhs_md, rhs_md, ret_md);
-      dnnl::matmul::primitive_desc pd(md, gemm_attr, g_comp->eng);
-      dnnl::primitive prim = dnnl::matmul(pd);
-      add_op(prim, {{DNNL_ARG_SRC, lhs->mem},
-                    {DNNL_ARG_WEIGHTS, rhs->mem},
-                    {DNNL_ARG_DST, ret_mem}});
-    } else
-      is_elements_add = true;
-  } else {
-    dnnl::matmul::desc md(lhs_md, rhs_md, ret_md);
-    dnnl::matmul::primitive_desc pd(md, g_comp->eng);
-    dnnl::primitive prim = dnnl::matmul(pd);
-    add_op(prim, {{DNNL_ARG_SRC, lhs->mem},
-                  {DNNL_ARG_WEIGHTS, rhs->mem},
-                  {DNNL_ARG_DST, ret_mem}});
-  }
-  odla_value v =
-      CreateValue(ret_mem, output_dims, is_elements_add ? nullptr : id);
-
-  InterpretIfNeeded();
-
-  return is_elements_add ? odla_Add(v, bias, id) : v;
-}
-
-odla_value batch_gemm_op(odla_value lhs, odla_bool transpose_lhs,
-                         odla_value rhs, odla_bool transpose_rhs,
-                         odla_float32 alpha, odla_float32 beta, odla_value bias,
-                         odla_value_shape output_dims, const odla_value_id id) {
+odla_value odla_Gemm(odla_value lhs, odla_bool transpose_lhs, odla_value rhs,
```

Comment:
after "gemm_op" is removed, can we still support "batch_gemm_op"?

Reply:
sure, i fused the batch_gemm and gemm function together